lib/build.ts

```diff
@@ -108,6 +108,12 @@ async function compileOnUnix(nodeVersion: string, targetArch: string) {
     args.push('--dest-cpu', cpu);
   }
 
+  if (hostArch !== targetArch) {
+    log.warn('Cross compiling!');
+    log.warn('You are responsible for appropriate env like CC, CC_host, etc.');
```

Comment:
This should link out to more information. What is CC, CC_host? Where can I learn more?

Reply:
Unfortunately there is no documentation there. Even Node.js does not provide one on the topic of cross-compilation. 

CC is "C Compiler".
CC_host is "C Compiler for the host". 

Users are kinda expected to know this, if they wish to take advantage of this more advanced option. Of course, I am always open to better docs, but I don't think I can do a "cross compile 101" here.

Reply:
maybe instead of making it working by default and show a warning would be better to force the flag to be there, if it is not there throw like it was before

Reply:
It would fail (very soon). So there is no need for that. That would also drastically increase the complexity of matrix-based CI workflows, as we probably don't want the flag when compile natively.

Reply:
ok LGTM for me so

Reply:
Let's add the documentation to the README then, and then link to the readme in the error message?

Reply:
and, FYI: upstream (Node.js) builds and tests natively. See https://github.com/nodejs/build . They have native machines donated by IBM, ARM, etc. That explains why there is very little documentation on cross compilation. 

There is no guaranteed good documentation here. Cross compilation is quite complicated. For instance, GCC vs LLVM: GCC uses separate binaries but LLVM uses `--target=<triple>` and other flags. And, the instructions for how to install a cross toolchain obviously vary a lot between distros. 

I think it would be challenging to add any documentation, and we should probably follow the example of upstream, and ask advanced users to explore the codes, workflows and Google for answers instead.